hosted-api/src/main/java/ai/vespa/hosted/api/ControllerHttpClient.java

```diff
@@ -118,4 +154,33 @@ private static Slime toSlime(byte[] data) {
         return new JsonDecoder().decode(new Slime(), data);
     }
 
+
+    /** Client that signs requests with a private key whose public part is assigned to an application in the remote controller. */
+    private static class SigningControllerHttpClient extends ControllerHttpClient {
+
+        private final RequestSigner signer;
+
+        private SigningControllerHttpClient(URI endpoint, Path privateKeyFile, ApplicationId id) {
+            super(endpoint, HttpClient.newBuilder());
+            this.signer = new RequestSigner(unchecked(() -> Files.readString(privateKeyFile, UTF_8)), id.serializedForm());
+        }
+
+        @Override
+        protected HttpRequest request(HttpRequest.Builder request, Method method, Supplier<InputStream> data) {
+            return signer.signed(request, method, data);
+        }
+
+    }
+
+
+    /** Client that uses a given Athenz identity to authenticate to the remote controller. */
```

Comment:
Do we need to leak the Athenz concept at all? Perhaps we can just call it `MutualTlsControllerHttpClient` or similar as there is nothing Athenz-specific with the client.

Reply:
I guess that's fail enough.